activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -750,7 +750,11 @@ def handle_warnings(sql)
           return if ActiveRecord.db_warnings_action.nil? || @raw_connection.warning_count == 0
 
           @affected_rows_before_warnings = @raw_connection.affected_rows
+          warning_count = @raw_connection.warning_count
           result = @raw_connection.query("SHOW WARNINGS")
+          result = [
+            [nil, nil, "Query had warning_count=#{warning_count} but ‘SHOW WARNINGS’ did not return the warnings. Check MySQL logs or database configuration."],
```

Comment:
Should not the level be "Warning"` instead of `nil`?

Reply:
Yup. We can. We aren't guaranteed that the warning that isn't surfaced is `"Warning"` but it's pretty likely.